spa/src/components/dashboard/sidebar/SidebarUpgradePrompt/SidebarUpgradePrompt.tsx

```diff
@@ -0,0 +1,80 @@
+import { Close } from '@material-ui/icons';
+import PropTypes, { InferProps } from 'prop-types';
+import { CloseButton, Header, UpgradeButton, Root, Text, IconWrapper } from './SidebarUpgradePrompt.styled';
+import ArrowUp from '@material-design-icons/svg/filled/keyboard_arrow_up.svg?react';
+import DoubleArrowUp from '@material-design-icons/svg/filled/keyboard_double_arrow_up.svg?react';
+import { SETTINGS } from 'routes';
+import { PLAN_NAMES } from 'constants/orgPlanConstants';
+import { HELP_URL } from 'constants/helperUrls';
+import { LinkProps } from 'react-router-dom';
+import { EnginePlan } from 'hooks/useContributionPage';
+import React from 'react';
+
+const SidebarUpgradePromptPropTypes = {
+  onClose: PropTypes.func.isRequired,
+  currentPlan: PropTypes.oneOf(Object.values(PLAN_NAMES)).isRequired
+};
+
+export type SidebarUpgradePromptProps = InferProps<typeof SidebarUpgradePromptPropTypes>;
+
+const COPY: Record<
+  EnginePlan['name'],
+  {
+    icon: React.FC<React.SVGProps<SVGSVGElement>>;
```

Comment:
```suggestion
    icon: React.ComponentType;
```

I think better to type it less restrictively in case we end up using a different approach on icons later.